### PR TITLE
Query modules can now answer in English

### PR DESCRIPTION
**Pull request by sveinbjornt**
* Query modules can now specify a voice id to use when synthesizing spoken audio answer
* Added English (UK) Abbi voice to azure voice module

### CODE REVIEW

File: queries/__init__.py [hunk 1]
```diff
@@ -339,6 +338,8 @@ def __init__(
         self.set_beautified_query(beautify_query(q))
         # Boolean flag for whether this is a voice query
         self._voice = voice
+        # Voice synthesizer ID, if any
+        self._voice_id = None
```
Comment (vthorsteinsson) [inline queries/__init__.py:342]: Add type annotation (cf. self._error below)?

File: queries/examples/english.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+def handle_plain_text(q: Query) -> bool:
+    ql = q.query_lower.rstrip("?")
+
+    if ql == "segðu eitthvað á ensku":
+        q.set_qtype("English")
+
+        # Set answer
+        answer = "This is an example of me speaking English"
+        q.set_answer(*gen_answer(answer))
+        q.set_voice_id("Abbi")
```
Comment (vthorsteinsson) [inline queries/examples/english.py:37]: Conert to q.set_voice_locale("en-US")

File: queries/__init__.py [hunk 3]
```diff
@@ -339,6 +338,10 @@ def __init__(
         self.set_beautified_query(beautify_query(q))
         # Boolean flag for whether this is a voice query
         self._voice = voice
+        # Voice synthesizer ID, if any
+        self._voice_id: Optional[str] = None
+        # Voice synthesizer locale
+        self._voice_locale: str = "is-IS"
```
Comment (vthorsteinsson) [inline queries/__init__.py:344]: Change to "is_IS" if we are sticking to underscores

File: speech/voices/__init__.py [hunk 4]
```diff
@@ -59,3 +59,20 @@ def generate_data_uri(data: bytes, mime_type: str = BINARY_MIMETYPE) -> str:
...
     b64str = b64encode(data).decode("ascii")
     return f"data:{mime_type};base64,{b64str}"
+
+
+# Map locales to a default voice ID
+LOCALE_TO_VOICE_ID = {
+    "is_IS": "Gudrun",
+    "en_US": "Jenny",
+    "en_GB": "Abbi",
+    "de_DE": "Amala",
```
Comment (vthorsteinsson) [inline speech/voices/__init__.py:69]: Add French, Danish, Swedish, Norwegian, Spanish, Polish

File: speech/voices/__init__.py [hunk 5]
```diff
@@ -59,3 +59,20 @@ def generate_data_uri(data: bytes, mime_type: str = BINARY_MIMETYPE) -> str:
...
+}
+assert "is_IS" in LOCALE_TO_VOICE_ID
+
+
+def voice_for_locale(locale: str) -> str:
+    """Returns default voice ID for the given locale. If locale is
+    not supported, returns the default voice ID for 'is_IS'."""
+    vid = LOCALE_TO_VOICE_ID.get(locale)
```
Comment (vthorsteinsson) [inline speech/voices/__init__.py:77]: Simpler: `return LOCALE_TO_VOICE_ID.get(locale, "is_IS")`

File: speech/voices/azure.py [hunk 6]
```diff
@@ -38,9 +38,22 @@
...
+    "Alfie": "en-GB-AlfieNeural",
+    # English (US)
+    "Jenny": "en-US-JennyNeural",
+    "Brandon": "en-US-BrandonNeural",
+    # German
+    "Amala": "de-DE-AmalaNeural",
+    # Many more voices available, see:
+    # https://learn.microsoft.com/en-us/azure/cognitive-services/speech-service/language-support
+}
+VOICES = _VOICE_TO_ID.keys()
```
Comment (vthorsteinsson) [inline speech/voices/azure.py:56]: `VOICES = frozenset(_VOICE_TO_ID.keys())`

File: speech/voices/azure.py [hunk 7]
```diff
@@ -38,9 +38,22 @@
...
+    # English (UK)
+    "Abbi": "en-GB-AbbiNeural",
+    "Alfie": "en-GB-AlfieNeural",
+    # English (US)
+    "Jenny": "en-US-JennyNeural",
+    "Brandon": "en-US-BrandonNeural",
+    # German
+    "Amala": "de-DE-AmalaNeural",
+    # Many more voices available, see:
+    # https://learn.microsoft.com/en-us/azure/cognitive-services/speech-service/language-support
```
Comment (vthorsteinsson) [inline speech/voices/azure.py:54]: Add languages here, cf. earlier comment

File: speech/voices/__init__.py (hunk 5 above)
Comment (sveinbjornt) [reply speech/voices/__init__.py:77]: This would return "is_IS", whereas the function is supposed to return a voice ID.

**Pull request merged**